contracts/LidoMatic.sol

```diff
@@ -0,0 +1,476 @@
+     */
+    function submit(uint256 _amount) external returns (uint256) {
+        require(_amount > 0, "Invalid amount");
+
+        IERC20(token).transferFrom(msg.sender, address(this), _amount);
+
+        uint256 totalShares = totalSupply();
+        uint256 totalPooledMatic = totalBuffered + totalDelegated;
+        uint256 amountToMint = totalShares != 0
+            ? (_amount * totalShares) / totalPooledMatic
+            : _amount;
+
+        _mint(msg.sender, amountToMint);
+
+        totalBuffered += _amount;
+
+        return amountToMint;
+    }
+
+    /**
+     * How to select a validator that we are going to withdraw from?
+     *
+     * @dev Stores users request to withdraw into a RequestWithdraw struct
+     * @param _amount - Amount of MATIC that is requested to withdraw
+     */
+    function requestWithdraw(uint256 _amount) external notPaused {
+        // Add a function that converts MATIC to StMATIC and reverse
+        // Check how many StMATIC does a caller have
+        // Add a mapping between a validator address and a nonce (starts at 1)
+        // Increment nonce by 1 everytime a user calls a withdrawal function
+        // Burn StMATIC after checking if _amount satisfies user's balance
+        // A nonce is dependent on a validator
+        // Add a nonce per user
+        Operator.OperatorShare[] memory operatorShares = nodeOperator
+            .getOperatorShares();
+
+        if (lastWithdrawnValidatorId > operatorShares.length - 1) {
+            lastWithdrawnValidatorId = 0;
+        }
+
+        address validatorShare = operatorShares[lastWithdrawnValidatorId]
+            .validatorShare;
+
+        uint256 callerBalance = balanceOf(msg.sender);
+
+        require(callerBalance >= _amount, "Invalid amount");
+
+        _burn(msg.sender, _amount);
+
+        uint256 amountInMATIC = getUserBalanceInMATIC();
+
+        sellVoucher_new(validatorShare, amountInMATIC, type(uint256).max);
+
+        if (validator2Nonce[validatorShare] == 0) {
+            validator2Nonce[validatorShare] = 1;
+        } else {
+            validator2Nonce[validatorShare]++;
+        }
+
+        if (user2Nonce[msg.sender] == 0) user2Nonce[msg.sender] = 1;
+
+        user2Nonce[msg.sender]++;
+
+        RequestWithdraw[] storage requestWithdraws = user2WithdrawRequest[
+            msg.sender
+        ];
+
+        requestWithdraws.push(
+            RequestWithdraw(
+                amountInMATIC,
+                validator2Nonce[validatorShare],
+                block.timestamp,
+                validatorShare,
+                true
+            )
+        );
+
+        lastWithdrawnValidatorId++;
+    }
+
+    /**
+     * @dev Delegates tokens to validator share contract
+     */
+    function delegate() external auth(DAO) {
+        Operator.OperatorShare[] memory operatorShares = nodeOperator
+            .getOperatorShares();
+
+        uint256 amountPerValidator = totalBuffered / operatorShares.length;
```

Comment:
I think this update will give us more clean numbers:
1. uint256 remainder = totalBuffered % operatorShares.length;
2. uint256 amountPerValidator = (totalBuffered - remainder) / operatorShares.length;

Reply:
I think it will be the same, because we are doing operations with integers :)